synapse/handlers/space_summary.py

```diff
@@ -141,12 +143,10 @@ async def get_space_summary(
                 # user is not permitted see.
                 #
                 # Filter the returned results to only what is accessible to the user.
-                room_ids = set()
                 events = []
-                for room in fed_rooms:
-                    fed_room_id = room.get("room_id")
-                    if not fed_room_id or not isinstance(fed_room_id, str):
-                        continue
```

Comment:
These checks were moved to `_summarize_remote_room` (since the `_RoomEntry` has a `room_id` attribute we need to check it before creating the object).